pkg/blackduck/reporting.go

```diff
@@ -42,48 +34,48 @@ func CreateSarifResultFile(vulns *Vulnerabilities, components *Components) *form
 	cweIdsForTaxonomies := []string{}
 	if vulns != nil && vulns.Items != nil {
 		for _, v := range vulns.Items {
-			component := componentLookup[fmt.Sprintf("%v/%v", v.Name, v.Version)]
-			result := *new(format.Results)
-			ruleId := v.Title()
-			log.Entry().Debugf("Transforming alert %v into SARIF format", ruleId)
-			result.RuleID = ruleId
+			result := format.Results{}
+			result.RuleID = v.VulnerabilityWithRemediation.VulnerabilityName
+			log.Entry().Debugf("Transforming alert %v on Package %v Version %v into SARIF format", result.RuleID, v.Component.Name, v.Component.Version)
 			result.Level = transformToLevel(v.VulnerabilityWithRemediation.Severity)
-			result.Message = new(format.Message)
+			result.Message = &format.Message{}
 			result.Message.Text = v.VulnerabilityWithRemediation.Description
-			result.AnalysisTarget = new(format.ArtifactLocation)
-			result.AnalysisTarget.URI = v.Name
+			result.AnalysisTarget = &format.ArtifactLocation{}
+			result.AnalysisTarget.URI = v.Component.ToPackageUrl().ToString()
 			result.AnalysisTarget.Index = 0
 			location := format.Location{PhysicalLocation: format.PhysicalLocation{ArtifactLocation: format.ArtifactLocation{URI: v.Name}}}
 			result.Locations = append(result.Locations, location)
-			partialFingerprints := new(format.PartialFingerprints)
-			partialFingerprints.PackageURLPlusCVEHash = base64.URLEncoding.EncodeToString([]byte(fmt.Sprintf("%v+%v", component.ToPackageUrl().ToString(), v.Title())))
-			result.PartialFingerprints = *partialFingerprints
-			cweIdsForTaxonomies = append(cweIdsForTaxonomies, v.VulnerabilityWithRemediation.CweID)
+			partialFingerprints := format.PartialFingerprints{}
```

Comment:
as above, could be simplified directly in this assignment:

result.PartialFingerprints = format.PartialFingerprints{
    PackageURLPlusCVEHash: ...,
}